### PR TITLE
Integrate the new kube-admission-webhook that generate certificates

### PR DESCRIPTION
**Pull request by qinqon**
The new version of kube-admission-webhook works at k8s and openshift and also handles server/ca certificate rotations.

It also includes some infra changes 
- Update kubevirtci
- Remove multistage build

```release-note
Use self generated certificate/key for webhook
```

### CODE REVIEW

File: config/default/manager/manager.yaml [hunk 1]
```diff
@@ -189,3 +199,13 @@ webhooks:
         apiGroups: ["kubevirt.io"]
         apiVersions: ["v1alpha3"]
         resources: ["virtualmachines"]
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: service
```
Comment (RamLavi) [inline config/default/manager/manager.yaml:206]: is this the secret added in the kube-addmissions? If so, then maybe the name is too general? what do you think?

**Comment by RamLavi** [comment]
Please add description to the PR and change the name of the PR

File: config/default/manager/manager.yaml (hunk 1 above)
Comment (RamLavi) [reply config/default/manager/manager.yaml:205]: forgot about kustomize adding the prefix. closing this..

**Comment by qinqon** [comment]
/retest

prow hiccup

**Comment by qinqon** [comment]
/retest

```
+ /tmp/kubemacpool/kubemacpool/_kubevirtci/cluster-up/up.sh
Download the image docker.io/kubevirtci/k8s-1.17@sha256:c3a307b55d60578d8aa99aca09d92728810d0ee6480887f8015ba21dd71d25b3
Downloading ...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Downloading ..............................................
Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:275: applying cgroup configuration for process caused \"mkdir /sys/fs/cgroup/memory/docker/bf4c7a447cba1a88a814843bca1518f2ebeeab419841beee23cece76af8b8759: cannot allocate memory\"": unknown
Makefile:91: recipe for target 'cluster-up' failed
make: *** [cluster-up] Error 1
+ EXIT_VALUE=2

```

**Comment by phoracek** [comment]
```
It also includes some infra changes

    Update kubevirtci
    Remove multistage build
```

Please drop everything that can be handled in a separate PR.

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,4 @@
-# Build the manager binary
-ARG GO_VERSION
```
Comment (phoracek) [inline Dockerfile:2]: This should be done in a separate PR

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# The path for kubevirtci installation
+_kubevirtci
```
Comment (phoracek) [inline .dockerignore:2]: Please use  a separate PR to add this change.

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+# The old path for kubevirtci installation, in case checkout is done from and
+# old branch this can exists
+kubevirtci
```
Comment (phoracek) [inline .dockerignore:6]: There was never such a directory in this project AFAIK

File: Makefile [hunk 5]
```diff
@@ -68,9 +67,14 @@ docker-goveralls: docker-builder docker-test
 docker-generate: docker-builder
 	DOCKER_BASE_IMAGE=${REGISTRY}/${IMG}:kubemacpool_builder ./hack/run.sh
 
-# Build the docker image
-docker-build: docker-builder
-	docker build . -t ${REGISTRY}/${IMG}:${IMAGE_TAG} --build-arg GO_VERSION=1.12.12
+docker-build: manager
```
Comment (phoracek) [inline Makefile:70]: Please drop the Makefile refactoring

File: config/default/manager/manager.yaml [hunk 6]
```diff
@@ -188,3 +198,13 @@ webhooks:
...
+  namespace: system
+type: kubernetes.io/tls
+data:
+  tls.crt: YmFkIGNlcnRpZmljYXRlCg==
```
Comment (phoracek) [inline config/default/manager/manager.yaml:209]: What do these values stand for?

File: pkg/manager/manager.go [hunk 7]
```diff
@@ -165,11 +164,6 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
 			return fmt.Errorf("unable to register webhooks to the manager error %v", err)
 		}
 
-		if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
```
Comment (phoracek) [inline pkg/manager/manager.go:168]: Is this change related?

Comment (qinqon) [reply pkg/manager/manager.go:168]: Yep, the HTTP has no use for us, it just say that the pod is up not that the webhook is ready, I have implement at "/readyz" at library so pod status will be ready when webhook is up and good it also checks that certs are good.

File: config/default/manager/manager.yaml (hunk 6 above)
Comment (qinqon) [reply config/default/manager/manager.yaml:209]:
```bash
$ echo "YmFkIGNlcnRpZmljYXRlCg==" | base64 -d
bad certificate
```

**Comment by qinqon** [comment]
/hold

This is using a kube-admission-webhook PR, let's unhold when it uses master.

**Comment by qinqon** [comment]
/hold cancel

using proper cert manager lib version

File: config/default/manager/manager.yaml (hunk 6 above)
Comment (ormergi) [reply config/default/manager/manager.yaml:209]: We need this in order to triger the manager to create new cert?
